src/Std/Data/Internal/List/Associative.lean

```diff
@@ -5242,6 +5242,68 @@ theorem getValueCast?_filter_containsKey [BEq α] [LawfulBEq α]
     case h_2 => rfl
     exact dl₁
 
+theorem getValueCast?_diff [BEq α] [LawfulBEq α]
```

Comment:
If you want to call if `diff` in your theorem names, consider adding a function `diff` that expands to `(List.filter (fun p => !containsKey p.fst l₂) l₁)`.